pkg/migrate/azure/privatedns.go

```diff
@@ -0,0 +1,744 @@
+package azure
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest/to"
+
+	"github.com/sirupsen/logrus"
+
+	aznetwork "github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-12-01/network"
+	azdns "github.com/Azure/azure-sdk-for-go/services/preview/dns/mgmt/2018-03-01-preview/dns"
+	azprivatedns "github.com/Azure/azure-sdk-for-go/services/privatedns/mgmt/2018-09-01/privatedns"
+	azconfig "github.com/openshift/installer/pkg/asset/installconfig/azure"
+)
+
+type legacyDNSZoneSerialized struct {
+	zone       []byte
+	recordsets [][]byte
+}
+
+type legacyDNSZoneInfo struct {
+	zone       *azdns.Zone
+	recordsets []*azdns.RecordSet
+}
+
+type legacyDNSClient struct {
+	resourceGroup    string
+	zonesClient      azdns.ZonesClient
+	recordsetsClient azdns.RecordSetsClient
+}
+
+func newLegacyDNSClient(session *azconfig.Session, resourceGroup string) *legacyDNSClient {
```

Comment:
this doesn't use resource group, so maybe drop it.

Reply:
the resource group is used/needed, so it's still here.